eng/packaging.targets

```diff
@@ -42,4 +54,76 @@
+        <RefAssemblyOutput>true</RefAssemblyOutput>
+        <PackagePath>ref/$([System.String]::new('%(ResolvedMatchingContract.SetTargetFramework)').Substring(16))</PackagePath>
+        <PackagePathWithFilename>ref/$([System.String]::new('%(ResolvedMatchingContract.SetTargetFramework)').Substring(16))/%(ResolvedMatchingContract.Filename).dll</PackagePathWithFilename>
+      </TfmSpecificPackageFile>
+      <TfmSpecificPackageFile Include="$(XmlDocFileRoot)1033\$(TargetName).xml" Condition="Exists('$(XmlDocFileRoot)1033\$(TargetName).xml')">
+        <RefAssemblyOutput>true</RefAssemblyOutput>
+        <PackagePath>ref/$([System.String]::new('%(ResolvedMatchingContract.SetTargetFramework)').Substring(16))</PackagePath>
+        <PackagePathWithFilename>ref/$([System.String]::new('%(ResolvedMatchingContract.SetTargetFramework)').Substring(16))/%(ResolvedMatchingContract.Filename).xml</PackagePathWithFilename>
+      </TfmSpecificPackageFile>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="AddRuntimeSpecificAssemblies"
+          DependsOnTargets="BuiltProjectOutputGroup;SatelliteDllsProjectOutputGroup;_AddPriFileToPackBuildOutput;$(TargetsForTfmSpecificBuildOutput)"
+          Condition="'$(TargetsAnyOS)' != 'true' and '$(ExcludeFromPackage)' == ''">
+    <ItemGroup>
+      <TfmSpecificPackageFile Include="@(SatelliteDllsProjectOutputGroupOutput);
+                            @(BuiltProjectOutputGroupOutput);
+                            @(_PathToPriFile)">
+        <PackagePath>runtimes/$(_runtimeOS)/lib/$(TargetFrameworkWithoutSuffix)</PackagePath>
+      </TfmSpecificPackageFile>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="AddRefDebugSymbolsWithTfm"
+          Condition="'$(IncludeSymbols)' == 'true' and '$(IncludeRefInPackage)' == 'true' and '$(TargetsAnyOS)' == 'true'">
+    <PropertyGroup>
+      <RefSymbolPath>$(BaseOutputPath)ref\$(TargetFrameworkWithoutSuffix)-$(Configuration)\$(TargetName).pdb</RefSymbolPath>
+    </PropertyGroup>
+
+    <ItemGroup Condition="Exists('$(RefSymbolPath)')">
+      <TfmSpecificDebugSymbolsFile Include="$(RefSymbolPath)">
+        <TargetPath>/ref/$(TargetFrameworkWithoutSuffix)/$(TargetName).pdb</TargetPath>
+        <TargetFramework>$(TargetFrameworkWithoutSuffix)</TargetFramework>
+      </TfmSpecificDebugSymbolsFile>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="AddRuntimeDebugSymbolsWithTfm"
+          Condition="'$(IncludeSymbols)' == 'true' and '$(TargetsAnyOS)' != 'true' and '$(ExcludeFromPackage)' == ''">
+    <PropertyGroup>
+      <RuntimeSymbolPath>$(BaseOutputPath)$(TargetFrameworkWithoutSuffix)-$(TargetFrameworkSuffix)-$(Configuration)\$(TargetName).pdb</RuntimeSymbolPath>
+    </PropertyGroup>
+
+    <ItemGroup Condition="Exists('$(RuntimeSymbolPath)')">
+      <TfmSpecificDebugSymbolsFile Include="$(RuntimeSymbolPath)">
+        <TargetPath>/runtimes/$(_runtimeOS)/lib/$(TargetFrameworkWithoutSuffix)/$(TargetName).pdb</TargetPath>
```

Comment:
How does that work for tfms like `net6.0-android`? Based on the property's name, I believe that `TargetFrameworkWithoutSuffix` never contains the rid part?

Reply:
_runtimeOS  = android in that case.